ghproxy/ghmetrics/ghpaths.go

```diff
@@ -0,0 +1,210 @@
+package ghmetrics
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	reposRegex          = regexp.MustCompile(`^/repos/(?P<owner>[^/]*)/(?P<repo>[^/]*)(?P<rest>/.*)$`)
+	varAndConstantRegex = regexp.MustCompile(`^/(?P<var>[^/]*)(?P<path>/.*)?$`)
+	constantAndVarRegex = regexp.MustCompile(`^/(?P<path>[^/]*)(?P<var>/.*)?$`)
+)
+
+func getSimplifiedPath(path string) string {
+	fragment := getFirstFragment(path)
+	switch fragment {
+	case "repos":
+		// /:owner/:repo
+		return handleRepos(path)
+	case "user":
+		// do some more
+		return path
+	case "users":
+		// do some more
+		return path
+	case "orgs":
+		// do some more
+		return path
+	case "issues":
+		// do some more
+		return path
+	case "search":
+		// do some more
+		return path
+	case "gists":
+		// do some more
+		return path
+	case "notifications":
+		// do some more
+		return path
+	case "repositories", "emojis", "events", "feeds", "hub", "rate_limits", "teams", "licenses":
+		return path
+	default:
+		logrus.WithField("path", path).Info("Path not handled")
+		return path
+	}
+}
+
+// getFirstFragment returns the first fragment of a path, of a
+// `*url.URL.Path`. e.g. `/repos/kubernetes/test-infra/` will return `repos`
+func getFirstFragment(path string) string {
```

Comment:
Could this use `strings.Split` from the stdlib?

Reply:
Will change this in an upcoming commit.

Reply:
Did you push this commit yet?